hooks/useRailgunProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import { networks } from "@/utils/networks";
+import { loadProviders } from "@/utils/railgun";
+import { setProviderForNetwork } from "@railgun-community/quickstart";
+import { useState, useEffect } from "react";
+import { useNetwork, useProvider } from "wagmi";
+
+export const useRailgunProvider = () => {
+  const [isProviderLoaded, setProviderLoaded] = useState<Boolean>(false);
+  const { chain } = useNetwork();
+  const network = networks[chain?.id || 1];
+  const provider = useProvider();
+
+  useEffect(() => {
+    const fn = async () => {
+      setProviderLoaded(false);
+      await loadProviders();
+      setProviderLoaded(true);
+    };
+    fn();
```

Comment:
Do we need a `.catch` or any error handling here, in case `loadProviders` errors?

Reply:
if `loadProviders` errors, i'd rather the error bubble up and throw at the top level